### PR TITLE
Updates to phing build process

### PR DESCRIPTION
**Pull request by initvector**
Updates to phing's build process

- [x] Better handling for version numbers
- [x] Better detection for build filename when cleaning up
- [x] Copy build files into new directory before zipping

### CODE REVIEW

**Comment by initvector** [comment]
Removing Debugger plug-in from build fixes https://github.com/vanilla/vanilla/issues/3262

File: build/build.xml [hunk 1]
```diff
@@ -8,21 +8,37 @@
...
     <target name="prepare">
-        <delete
-            file="build.zip" />
+        <loadfile property="vanilla.index" file="${project.dir}/index.php"/>
+
+        <!-- Grab the version number from Vanilla's index.php. -->
         <propertyregex property="vanilla.version"
             subject="${vanilla.index}"
-            pattern="define\('APPLICATION_VERSION', '(\d+\.\d+\.\d+)'\)"
+            pattern="define\('APPLICATION_VERSION', '([\d\.]+)'\)"
```
Comment (linc) [inline build/build.xml]: This needs to allow for `a`, `b`, `rc` & `p` as well.

**Comment by linc** [comment]
Aside from the version matching, this looks good.

**Comment by initvector** [comment]
Rebasing this has done something to the history that makes me nervous.  Closing in favor of opening a fresh, new PR.

**Pull request closed without merging**